clang/lib/Serialization/ASTWriter.cpp

```diff
@@ -6659,13 +6655,22 @@ void ASTWriter::MacroRead(serialization::MacroID ID, MacroInfo *MI) {
 }
 
 void ASTWriter::TypeRead(TypeIdx Idx, QualType T) {
-  // Always take the highest-numbered type index. This copes with an interesting
+  // Always take the type index that comes in later module files.
+  // This copes with an interesting
   // case for chained AST writing where we schedule writing the type and then,
   // later, deserialize the type from another AST. In this case, we want to
-  // keep the higher-numbered entry so that we can properly write it out to
+  // keep the just writing entry so that we can properly write it out to
```

Comment:
```suggestion
  // keep the entry from a later module so that we can properly write it out to
```

NIT: the "just writing" sounds weird and there is a reference to the fact it's being written later in the line anyway.

Reply:
Done